server/coordinator/entry_lock.go

```diff
@@ -0,0 +1,62 @@
+package coordinator
+
+import (
+	"sort"
+	"sync"
+)
+
+type EntryLock interface {
+	TryLock(locks []uint64) bool
+	ReleaseLock(locks []uint64) bool
+}
+
+type EntryLockImpl struct {
+	lock       sync.Mutex
+	entryLocks map[uint64]bool
+}
+
+func NewEntryLock() EntryLock {
+	return &EntryLockImpl{
+		entryLocks: make(map[uint64]bool, 0),
+	}
+}
+
+func (l *EntryLockImpl) TryLock(locks []uint64) bool {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	sort.Slice(locks, func(i, j int) bool {
+		return i > j
+	})
+
+	for _, lock := range locks {
+		_, exists := l.entryLocks[lock]
+		if exists {
+			return false
+		}
+	}
+
+	for _, lock := range locks {
+		l.entryLocks[lock] = true
+	}
+
+	return true
+}
+
+func (l *EntryLockImpl) ReleaseLock(locks []uint64) bool {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	for _, lock := range locks {
+		_, exists := l.entryLocks[lock]
+		if !exists {
+			return false
+		}
```

Comment:
A better behavior is just panic if the caller unlocks a wrong batch, and no need to return a boolean to tell the caller whether to succeed in unlocking.